Tauri V2 - Wallet Migration

tommyv1987 opened the pull request:
- The core of the lifting was done via the migrate command
- A lot of API's changed
- Some help from SDK (Andrew) this is always welcomed
- Improved styling
- Pipelines are fixed for all platforms
- **TauriV2 is here**

A lot of pieces were refactored ☝️ 

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/nymtech/nym/5687)
<!-- Reviewable:end -->

tommyv1987 committed:
migrate to v2

- lots to check and do

tommyv1987 committed:
migrate to v2

- lots to check and do

tommyv1987 committed:
migrate to v2

- lots to check and do

tommyv1987 committed:
migrate to v2

- lots to check and do

tommyv1987 committed:
add the base points back in

 - now i've reverted back to the original two here, as the compiler is failing around `tauri::api::path` however, looking into the new design for the path resolver in tower this tasks, requires pratically changing the whole wallet_strorage and config set up
- it seems pretty straight forward https://v2.tauri.app/start/migrate/from-tauri-1/#migrate-path-to-tauri-manager - however, I would need a second set of eyes on this

tommyv1987 committed:
add the base points back in

 - now i've reverted back to the original two here, as the compiler is failing around `tauri::api::path` however, looking into the new design for the path resolver in tower this tasks, requires pratically changing the whole wallet_strorage and config set up
- it seems pretty straight forward https://v2.tauri.app/start/migrate/from-tauri-1/#migrate-path-to-tauri-manager - however, I would need a second set of eyes on this

tommyv1987 committed:
add the base points back in

 - now i've reverted back to the original two here, as the compiler is failing around `tauri::api::path` however, looking into the new design for the path resolver in tower this tasks, requires pratically changing the whole wallet_strorage and config set up
- it seems pretty straight forward https://v2.tauri.app/start/migrate/from-tauri-1/#migrate-path-to-tauri-manager - however, I would need a second set of eyes on this

tommyv1987 committed:
Merge remote-tracking branch 'origin/feature/test-v2' into feature/test-v2

tommyv1987 committed:
Merge remote-tracking branch 'origin/feature/test-v2' into feature/test-v2

tommyv1987 committed:
Merge remote-tracking branch 'origin/feature/test-v2' into feature/test-v2

tommyv1987 committed:
Revert "add the base points back in"

This reverts commit 400aa6ba6d638d0a7e4a062936e883a0a03a31b8.

tommyv1987 committed:
update to log in

- next up fix hyperlinks

tommyv1987 committed:
- Update beyond tauri v2

- use the latest and greatest
- fixed links to use the command shell
- app version changes, need to be fixed to allow the auto updater too work

tommyv1987 committed:
More permissions errors

- fix more perm errors
- enabled the version in the wallet

tommyv1987 committed:
Add pruning warning errors

tommyv1987 committed:
Rename and update

tommyv1987 committed:
Init clipboard manager

tommyv1987 committed:
Remove legacy

tommyv1987 committed:
Jazz up receive modal

tommyv1987 committed:
Input fields

tommyv1987 committed:
Update account display

tommyv1987 committed:
attempt to push to ci

tommyv1987 committed:
Update runner for linux

tommyv1987 committed:
update runner platform

tommyv1987 committed:
fix app deps

tommyv1987 committed:
test old runner first

tommyv1987 committed:
try 22.04

tommyv1987 committed:
attempt windows build

tommyv1987 committed:
correct tauri path

tommyv1987 committed:
tauri path

tommyv1987 committed:
update windows build

tommyv1987 committed:
revert previous add more logging

tommyv1987 committed:
another attempt

tommyv1987 committed:
format

tommyv1987 committed:
try again

tommyv1987 committed:
remove arg

tommyv1987 committed:
Fixing all yarn lint errors

tommyv1987 committed:
No need to publish these to the build server just use the artifacts

tommyv1987 committed:
bump to version 2.0.0

- it's a big release therefore let's semver it correctly

tommyv1987 committed:
fix all workflows

tommyv1987 committed:
Merge branch 'develop' into feature/test-v2

Commit:
merge conflicts

tommyv1987 committed:
removed duplicates and reverted back to 1.2.18 as a version

doums commented on nym-wallet/package.json:
```diff
@@ -35,7 +36,11 @@
     "@storybook/react": "^6.5.15",
-    "@tauri-apps/api": "^1.2.0",
+    "@tauri-apps/api": "^2.4.0",
+    "@tauri-apps/cli": "^2.4.0",
```
`@tauri-apps/cli` should be defined as dev-dependency only

tommyv1987 replied:
Fixed 👍

doums commented on nym-wallet/src-tauri/Cargo.toml:
```diff
@@ -1,25 +1,29 @@
-tauri-codegen = "=1.2.1"
-tauri-macros = "=1.2.1"
+tauri-codegen = "=2.1.1"
+tauri-macros = "2.1.1"
```
Are these 2 deps still needed?

tommyv1987 replied:
Removed, you're right they're not needed 👍

doums commented on nym-wallet/src-tauri/capabilities/main-capability.json:
```diff
@@ -0,0 +1,31 @@
+        "core:resources:default",
+        "core:menu:default",
+        "core:tray:default",
+        "shell:allow-open",
```
Suggestion: for opener it is encouraged to switch on https://tauri.app/plugin/opener/ (and drop shell)

tommyv1987 replied:
Dropped this too

tommyv1987 committed:
fix delegations page - after overflow

doums commented on nym-wallet/src-tauri/src/operations/app/link.rs:
```diff
@@ -0,0 +1,22 @@
+        "windows" => Command::new("cmd").args(["/c", "start", &url]).status(),
+        "linux" => Command::new("xdg-open").arg(&url).status(),
+        os => return Err(format!("Unsupported OS: {}", os)),
+    };
```
yea instead of this manual implem you should use the opener plugin https://tauri.app/plugin/opener/

tommyv1987 replied:
Now using the plugin

doums commented on nym-wallet/src/components/Send/SendInputModal.tsx:
```diff
@@ -52,28 +65,124 @@ export const SendInputModal = ({
+    } else {
+      setNoAccount(false);
+    }
+  }, [balance]);
```
With such simple state/props conditions, it is better to avoid unnecessary `useEffect`:
```
// component root scope
const noAccount = !balance || balance === '0' || parseFloat(balance) === 0;
```

tommyv1987 replied:
Nice one, changed this as well

doums commented on nym-wallet/src/components/TauriLinkWrapper.tsx:
```diff
@@ -0,0 +1,21 @@
+    if (href && (href.startsWith('http://') || href.startsWith('https://'))) {
+      event.preventDefault();
+      console.log('Opening link in browser:', href);
+      await openInBrowser(href);
```
suggestion: you can avoid custom implem when using plugin-opener

```
import { openUrl } from '@tauri-apps/plugin-opener';
```

tommyv1987 replied:
Replaced this with the `plugin-opener` :)

doums commented on nym-wallet/src/theme/NymWalletTheme.tsx:
```diff
@@ -1,17 +1,38 @@
+    // Add Inter font for modern typography
+    const interFontLink = document.createElement('link');
+    interFontLink.rel = 'stylesheet';
+    interFontLink.href = 'https://fonts.googleapis.com/css2?family=Inter:wght@300;400;500;600;700&display=swap';
```
This should be setup before entering react app, ie. in the JS entry point
Also note that when running in development react hooks run twice for stress purposes.
At least, you should protect it against re-runs

```
import [...]

// outside react tree
let init = false;

const FontLoader = () => {
  useEffect(() => {
    if (init) { return; }
    init = true;
    // logic
```

tommyv1987 replied:
Noted and changed

fmtabbara replied:
I'm not sure that this is the correct way to set the fonts up. If it's working leave it like this but we may want revisit - https://mui.com/material-ui/react-typography

doums commented:
In the `Cargo.toml` I'm not sure if this feature is still needed
```
custom-protocol = ["tauri/custom-protocol"]
```

tommyv1987 committed:
Fix delegation list

tommyv1987 committed:
first code review comments

tommyv1987 committed:
Fix PR comments

- removed the shell open in favour of the tauri plugin for opening
- cleaned up some code
- added a few packages

tommyv1987 committed:
paste not working from currency form

- removed shellhelper too

tommyv1987 committed:
yarn lint fix

tommyv1987 committed:
cargo fmt

- hopefully the last

tommyv1987 committed:
fix lint again

fmtabbara commented:
Generally looks OK. A couple things that stand out from a frontend perspective are

1. There are quite a lot of inline styles now - I think this is OK for now and necessary to get the new theme in quickly. If/when there's time in the future we should update the global theme in order to reduce the number of inline styles.

2. The vesting feature is still there - Can this be removed yet?

tommyv1987 committed:
fix wallet ci

tommyv1987 commented:
> Generally looks OK. A couple things that stand out from a frontend perspective are
> 
> 1. There are quite a lot of inline styles now - I think this is OK for now and necessary to get the new theme in quickly. If/when there's time in the future we should update the global theme in order to reduce the number of inline styles.
> 2. The vesting feature is still there - Can this be removed yet?

Unfortunately we can't remove vesting just yet @fmtabbara, otherwise this would've been the first thing to remove :) 

As for styling let's look at this after as well

tommyv1987 commented:
@fmtabbara I haven't had the chance to update the storybook components, but that too, can come in the future

tommyv1987 committed:
fix remaining lint and cargo clippy errors

tommyv1987 committed:
conflicts
